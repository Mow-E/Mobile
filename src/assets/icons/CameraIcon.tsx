import * as React from 'react';
import Svg, {SvgProps, Rect, Circle, Line, Path} from 'react-native-svg';
import {ColorValue} from 'react-native';
import IconProps from './IconProps';
import colors from '../../styles/colors';

/**
 * The (extended) properties for the camera icon.
 */
type SvgComponentProps = Omit<SvgProps, 'fill'> & {
  /**
   * The primary color of the icon, used for the main part of the camera.
   */
  primaryColor: ColorValue;
  /**
   * The secondary color of the icon, used for the lens section of the camera.
   */
  secondaryColor: ColorValue;
  /**
   * The color of the lens in the icon.
   */
  lensColor: ColorValue;
};

const SvgComponent = (props: SvgComponentProps) => (
  <Svg width={24} height={24} viewBox="0 0 88 57.878315" fill="none" {...props}>
    <Rect
      x={0.5}
      y={8.9748135}
      width={87}
      height={48.4035}
      fill={props.primaryColor}
      stroke={props.stroke}
    />
    <Circle
      cx={54.421101}
      cy={33.562572}
      r={15.3246}
      fill={props.secondaryColor}
      stroke={props.stroke}
    />
    <Circle
      cx={54.806999}
      cy={33.948772}
      r={11.8509}
      fill={props.lensColor}
      stroke={props.stroke}
    />
    <Line
      y1={20.325872}
      x2={46.3158}
      y2={20.325872}
      stroke={props.stroke}
      x1={0}
    />
    <Line
      y1={46.571472}
      x2={46.3158}
      y2={46.571472}
      stroke={props.stroke}
      x1={0}
    />
    <Line
      x1={62.526299}
      y1={46.571472}
      x2={88}
      y2={46.571472}
      stroke={props.stroke}
    />
    <Line
      x1={62.526299}
      y1={20.325872}
      x2={88}
      y2={20.325872}
      stroke={props.stroke}
    />
    <Path
      d="m 50.1754,0.75557355 -0.3779,0.43643995 -6.0166,6.94736 -0.7165,0.82733 H 70.0144 l -0.3779,-0.43643995 -6.0166,-6.94736 -0.7165,-0.82733 Z"
      fill={props.primaryColor}
    />
  </Svg>
);

/**
 * An icon that shows a camera.
 * Used as the main icon for the image history section(s).
 */
function CameraIcon({
  size = 24,
  colored = false,
  darkModeInverted = false,
}: IconProps): JSX.Element {
  const darkModeStrokeColor = colored ? colors.gray['50'] : colors.gray['450'];
  const strokeColor = darkModeInverted
    ? darkModeStrokeColor
    : colors.gray['950'];

  const primaryColoredFillColor = colors.primary.dark;
  const primaryGrayscaleFillColor = colors.gray['300'];
  const primaryFillColor = colored
    ? primaryColoredFillColor
    : primaryGrayscaleFillColor;

  const secondaryColoredFillColor = colors.primary.light;
  const secondaryGrayscaleFillColor = colors.gray['200'];
  const secondaryFillColor = colored
    ? secondaryColoredFillColor
    : secondaryGrayscaleFillColor;

  const lensColor =
    colored && darkModeInverted ? colors.gray['700'] : colors.gray['50'];

  return (
    <SvgComponent
      height={size}
      width={size}
      stroke={strokeColor}
      primaryColor={primaryFillColor}
      secondaryColor={secondaryFillColor}
      lensColor={lensColor}
      strokeWidth={size > 128 ? 128 / size : 24 / size}
    />
  );
}

export default CameraIcon;
