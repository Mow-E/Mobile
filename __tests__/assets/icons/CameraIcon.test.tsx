import 'react-native';
import React from 'react';
import CameraIcon from '../../../src/assets/icons/CameraIcon';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders default correctly', () => {
  const tree = renderer.create(<CameraIcon />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders colored correctly', () => {
  const tree = renderer.create(<CameraIcon colored />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders uncolored correctly', () => {
  const tree = renderer.create(<CameraIcon colored={false} />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders in different sizes', () => {
  const treeSmall = renderer.create(<CameraIcon size={24} />).toJSON();
  const treeLarge = renderer.create(<CameraIcon size={48} />).toJSON();

  expect(treeSmall).not.toEqual(treeLarge);
});
