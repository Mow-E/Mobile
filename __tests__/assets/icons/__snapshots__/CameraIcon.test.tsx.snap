// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders colored correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  fill="none"
  focusable={false}
  height={24}
  lensColor="#ffffff"
  meetOrSlice={0}
  minX={0}
  minY={0}
  primaryColor="#24B655"
  secondaryColor="#8DDFA9"
  stroke="#000000"
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={57.878315}
  vbWidth={88}
  width={24}
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
        "stroke",
      ]
    }
    stroke={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGRect
      fill={
        {
          "payload": 4280596053,
          "type": 0,
        }
      }
      height="48.4035"
      propList={
        [
          "fill",
          "stroke",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      width="87"
      x="0.5"
      y="8.9748135"
    />
    <RNSVGCircle
      cx="54.421101"
      cy="33.562572"
      fill={
        {
          "payload": 4287487913,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
          "stroke",
        ]
      }
      r="15.3246"
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    />
    <RNSVGCircle
      cx="54.806999"
      cy="33.948772"
      fill={
        {
          "payload": 4294967295,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
          "stroke",
        ]
      }
      r="11.8509"
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    />
    <RNSVGLine
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "stroke",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      x1="0"
      x2="46.3158"
      y1="20.325872"
      y2="20.325872"
    />
    <RNSVGLine
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "stroke",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      x1="0"
      x2="46.3158"
      y1="46.571472"
      y2="46.571472"
    />
    <RNSVGLine
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "stroke",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      x1="62.526299"
      x2="88"
      y1="46.571472"
      y2="46.571472"
    />
    <RNSVGLine
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "stroke",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      x1="62.526299"
      x2="88"
      y1="20.325872"
      y2="20.325872"
    />
    <RNSVGPath
      d="m 50.1754,0.75557355 -0.3779,0.43643995 -6.0166,6.94736 -0.7165,0.82733 H 70.0144 l -0.3779,-0.43643995 -6.0166,-6.94736 -0.7165,-0.82733 Z"
      fill={
        {
          "payload": 4280596053,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`renders default correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  fill="none"
  focusable={false}
  height={24}
  lensColor="#ffffff"
  meetOrSlice={0}
  minX={0}
  minY={0}
  primaryColor="#9d9da2"
  secondaryColor="#d9d9d9"
  stroke="#000000"
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={57.878315}
  vbWidth={88}
  width={24}
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
        "stroke",
      ]
    }
    stroke={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGRect
      fill={
        {
          "payload": 4288519586,
          "type": 0,
        }
      }
      height="48.4035"
      propList={
        [
          "fill",
          "stroke",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      width="87"
      x="0.5"
      y="8.9748135"
    />
    <RNSVGCircle
      cx="54.421101"
      cy="33.562572"
      fill={
        {
          "payload": 4292467161,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
          "stroke",
        ]
      }
      r="15.3246"
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    />
    <RNSVGCircle
      cx="54.806999"
      cy="33.948772"
      fill={
        {
          "payload": 4294967295,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
          "stroke",
        ]
      }
      r="11.8509"
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    />
    <RNSVGLine
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "stroke",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      x1="0"
      x2="46.3158"
      y1="20.325872"
      y2="20.325872"
    />
    <RNSVGLine
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "stroke",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      x1="0"
      x2="46.3158"
      y1="46.571472"
      y2="46.571472"
    />
    <RNSVGLine
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "stroke",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      x1="62.526299"
      x2="88"
      y1="46.571472"
      y2="46.571472"
    />
    <RNSVGLine
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "stroke",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      x1="62.526299"
      x2="88"
      y1="20.325872"
      y2="20.325872"
    />
    <RNSVGPath
      d="m 50.1754,0.75557355 -0.3779,0.43643995 -6.0166,6.94736 -0.7165,0.82733 H 70.0144 l -0.3779,-0.43643995 -6.0166,-6.94736 -0.7165,-0.82733 Z"
      fill={
        {
          "payload": 4288519586,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`renders uncolored correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  fill="none"
  focusable={false}
  height={24}
  lensColor="#ffffff"
  meetOrSlice={0}
  minX={0}
  minY={0}
  primaryColor="#9d9da2"
  secondaryColor="#d9d9d9"
  stroke="#000000"
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={57.878315}
  vbWidth={88}
  width={24}
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
        "stroke",
      ]
    }
    stroke={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGRect
      fill={
        {
          "payload": 4288519586,
          "type": 0,
        }
      }
      height="48.4035"
      propList={
        [
          "fill",
          "stroke",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      width="87"
      x="0.5"
      y="8.9748135"
    />
    <RNSVGCircle
      cx="54.421101"
      cy="33.562572"
      fill={
        {
          "payload": 4292467161,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
          "stroke",
        ]
      }
      r="15.3246"
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    />
    <RNSVGCircle
      cx="54.806999"
      cy="33.948772"
      fill={
        {
          "payload": 4294967295,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
          "stroke",
        ]
      }
      r="11.8509"
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    />
    <RNSVGLine
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "stroke",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      x1="0"
      x2="46.3158"
      y1="20.325872"
      y2="20.325872"
    />
    <RNSVGLine
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "stroke",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      x1="0"
      x2="46.3158"
      y1="46.571472"
      y2="46.571472"
    />
    <RNSVGLine
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "stroke",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      x1="62.526299"
      x2="88"
      y1="46.571472"
      y2="46.571472"
    />
    <RNSVGLine
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "stroke",
        ]
      }
      stroke={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      x1="62.526299"
      x2="88"
      y1="20.325872"
      y2="20.325872"
    />
    <RNSVGPath
      d="m 50.1754,0.75557355 -0.3779,0.43643995 -6.0166,6.94736 -0.7165,0.82733 H 70.0144 l -0.3779,-0.43643995 -6.0166,-6.94736 -0.7165,-0.82733 Z"
      fill={
        {
          "payload": 4288519586,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
